File: repos_estudiantes_ALVAREZ_primera_entrega/ILLANES_VASQUEZ_GABRIEL/IntroduccionAPOO/EJERCICIO3/Coche.py
class Coche:
    def __init__(self, marca, modelo):
        self.marca = marca
        self.modelo = modelo
        self.velocidad = 0

    def acelerar(self):
        self.velocidad += 10
        print("El coche", self.marca, self.modelo, "aceleró:", self.velocidad, "km/h")

    def frenar(self):
        if self.velocidad >= 5:
            self.velocidad -= 5
        else:
            self.velocidad = 0
        print("El coche", self.marca, self.modelo, "frenó:", self.velocidad, "km/h")


if __name__ == "__main__":
    c1 = Coche("Toyota", "Corolla")
    c2 = Coche("Ford", "Fiesta")

    c1.acelerar()
    c1.frenar()

    c2.acelerar()
    c2.acelerar()
    c2.frenar()

    print("Velocidad final del", c1.marca, c1.modelo + ":", c1.velocidad, "km/h")
    print("Velocidad final del", c2.marca, c2.modelo + ":", c2.velocidad, "km/h")

File: repos_estudiantes_ALVAREZ_primera_entrega/ILLANES_VASQUEZ_GABRIEL/IntroduccionAPOO/EJERCICIO3/Coche.java

package ejercicio7;



public class Coche {
    public String marca;
    public String modelo;
    public int velocidad;

    public Coche(String marca, String modelo) {
        this.marca = marca;
        this.modelo = modelo;
        this.velocidad = 0; 
    }

    public void acelerar() {
        this.velocidad += 10; 
        System.out.println("El coche " + marca + " " + modelo + " aceleró: " + velocidad + " km/h");
    }

    public void frenar() {
        if (this.velocidad >= 5) {
            this.velocidad -= 5; 
        } else {
            this.velocidad = 0; 
        }
        System.out.println("El coche " + marca + " " + modelo + " frenó: " + velocidad + " km/h");
    }

    public static void main(String[] args) {

        Coche c1 = new Coche("Toyota", "Corolla");
        Coche c2 = new Coche("Ford", "Fiesta");

        c1.acelerar();
        c1.frenar();

        c2.acelerar();
        c2.acelerar();
        c2.frenar();

        System.out.println("Velocidad final del " + c1.marca + " " + c1.modelo + ": " + c1.velocidad + " km/h");
        System.out.println("Velocidad final del " + c2.marca + " " + c2.modelo + ": " + c2.velocidad + " km/h");
    }
}

